<a href="https://colab.research.google.com/github/CakeNuthep/Super_AI/blob/master/StatisticalHypothesis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# สมมุติฐาน
H0 = ทั้งสองวิธีได้ยอดขายเท่ากัน (D0 = 0)

H1 = ยอดขายที่วางสินค้าระดับชั้นสายตาสูงกว่าที่ระดับต่ำกว่าสายตา (D0 > 0)

กำหนดให้ Alpha = 0.01

In [1]:
from scipy import stats
import numpy as np

In [2]:
eye_level = [181,172,190,187,210,202,166,173,183,184]
bellow_eye_level = [178,172,185,184,201,201,160,168,180,179] 


In [3]:
# cal diff
eye_level = np.array(eye_level)
bellow_eye_level = np.array(bellow_eye_level)

diff_data = np.subtract(eye_level,bellow_eye_level)
diff_data = np.reshape(diff_data,(len(diff_data),1))
diff_data

array([[3],
       [0],
       [5],
       [3],
       [9],
       [1],
       [6],
       [5],
       [3],
       [5]])

In [8]:
# 1. Set assumption: D0=0.0 vs H1: D0 > 0 (One-tail testing)
Mu_D0=0
tail=1 # Valid value of tail are -1,1,2

# 2. Set level of significance:
alpha = 0.01

# 3. Select a statistical test: t-test
tval,pval =stats.ttest_1samp(diff_data,Mu_D0)

# 4. Draw the conclusion based on p-value.
pval=pval/2.0 
if tail==-1.0:
    print("p-value: %f" % pval)
    if (pval<alpha and tval <0) :
        print(" Reject the null hypothesis (H0)")
    else:
        print(" Fail to reject the null hypothesis (H0)")
else : 
    print("p-value: %f" % pval)
    if (pval<alpha and tval > 0) :
        print(" Reject the null hypothesis (H0)")
    else:
        print(" Fail to reject the null hypothesis (H0)")

p-value: 0.000424
 Reject the null hypothesis (H0)


# จากผลที่ได้

ค่า p-value มีค่าเป็น 0.000424 ซึ่งมีค่าน้อยกว่า alpha ดังนั้น จึงเป็นการปฎิเสธสมมุติฐานหลัก หรือก็คือ  ยอดขายที่วางสินค้าระดับชั้นสายตามียอดขายสูงกว่าที่ระดับต่ำกว่าสายตา